### DISCUSSION TITLE
How can I load asset with label and settings?

### DISCUSSION THREAD

**Discussion by Jly13** [bevyengine/bevy#12258] [Q&A]
When I try to load the asset with label and settings, the settings are always ignored.
```rust
let scene:Handle<Scene> = asset_server.load_with_settings("models/FlightHelmet/FlightHelmet.gltf#Scene0", |s: &mut bevy::gltf::GltfLoaderSettings| {
        error!("This has never been output!");
        s.include_source = true;
    });
```
Is this reasonable? What can I do when I need both label and settings?

**Comment by rcorre** [comment]
`include_source` sets the `source` property on the `Gltf` asset, so it doesn't make sense to pass if you're instantiating a Scene as a bevy type. How would you access the `source`? You'd need to load the `.gltf` without a label, then access the scene on the gltf asset as described [in the cheatbook](https://bevy-cheatbook.github.io/3d/gltf.html#gltf-master-asset). What are you trying to achieve?

**Comment by Jly13** [comment]
I picked this example just to explain the issue and make it simpler to test. Thank you for your reply.



---- Replied Message ----
| From | Ryan ***@***.***> |
| Date | 04/19/2024 20:00 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [bevyengine/bevy] How can I load asset with label and settings? (Discussion #12258) |

include_source sets the source property on the Gltf asset, so it doesn't make sense to pass if you're instantiating a Scene as a bevy type. How would you access the source? You'd need to load the .gltf without a label, then access the scene on the gltf asset as described in the cheatbook. What are you trying to achieve?
